test_agent/rust/src/utils.rs

```diff
@@ -0,0 +1,552 @@
+        _uattributes.source = MessageField(Some(Box::new(_source.0)));
+        }
+        if!(_sink.0.clone() == UUri::default()){
+            _uattributes.sink = MessageField(Some(Box::new(_sink.0)));
+            }
+            _uattributes.priority = _priority.into();
+            _uattributes.ttl = _ttl.into();
+            _uattributes.permission_level = Some(_permission_level.into());
+            _uattributes.commstatus = Some(_commstatus.into());
+            _uattributes.reqid = __reqid;
+            _uattributes.token= Some(_token.to_owned());
+            _uattributes.traceparent = Some(_traceparent.to_owned());
+        
+        Ok(WrapperUAttribute( _uattributes))
+    }
+}
+#[derive(Default)]
+pub struct WrapperUPayload(pub UPayload);
+impl<'de> Deserialize<'de> for WrapperUPayload {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let value: Value = Deserialize::deserialize(deserializer)?;
+
+        fn from_str_format(s: &str) -> UPayloadFormat {
+            match s {
+                "UPAYLOAD_FORMAT_JSON" => UPayloadFormat::UPAYLOAD_FORMAT_JSON,
+                "UPAYLOAD_FORMAT_PROTOBUF" => UPayloadFormat::UPAYLOAD_FORMAT_PROTOBUF,
+                "UPAYLOAD_FORMAT_RAW" => UPayloadFormat::UPAYLOAD_FORMAT_RAW,
+                "UPAYLOAD_FORMAT_SOMEIP" => UPayloadFormat::UPAYLOAD_FORMAT_SOMEIP,
+
+                "UPAYLOAD_FORMAT_PROTOBUF_WRAPPED_IN_ANY" => {
+                    UPayloadFormat::UPAYLOAD_FORMAT_PROTOBUF_WRAPPED_IN_ANY
+                }
+                "UPAYLOAD_FORMAT_SOMEIP_TLV" => UPayloadFormat::UPAYLOAD_FORMAT_SOMEIP_TLV,
+                "UPAYLOAD_FORMAT_TEXT" => UPayloadFormat::UPAYLOAD_FORMAT_TEXT,
+                "UPAYLOAD_FORMAT_UNSPECIFIED" => UPayloadFormat::UPAYLOAD_FORMAT_UNSPECIFIED,
+                _ => UPayloadFormat::UPAYLOAD_FORMAT_UNSPECIFIED,
+            }
+        }
+
+        let _format = match value.get("format") {
+            Some(_format) => from_str_format(
+                _format
+                    .as_str()
+                    .expect("Deserialize:something wrong with _type field"),
+            ),
+            None => UPayloadFormat::UPAYLOAD_FORMAT_UNSPECIFIED,
+        };
+
+        let _length = match value.get("length") {
+            Some(_length) => _length
+                .as_str()
+                .unwrap_or_else(|| panic!("Deserialize: something wrong with commstatus field"))
+                .parse::<i32>()
+                .expect("commstatus parsing error"),
+            None => 0,
+        };
+
+        
+         
+        let _data = match value.get("value") {
+            Some(_data) => Data::Value( serde_json::to_vec(_data).expect("error in converting data value to vector")),
+            None => Data::Reference(0),
+        };
+
+        let _special_fields = SpecialFields::default();
+
+        Ok(WrapperUPayload(UPayload {
```

Comment:
Same sorts of suggestions as [here](https://github.com/eclipse-uprotocol/up-tck/pull/23/files#r1566193463).

Reply:
corrected